internal/webhook/template_webhook.go

```diff
@@ -55,7 +62,24 @@ func (*ClusterTemplateValidator) ValidateUpdate(_ context.Context, _ runtime.Obj
 }
 
 // ValidateDelete implements webhook.Validator so a webhook will be registered for the type.
-func (*ClusterTemplateValidator) ValidateDelete(_ context.Context, _ runtime.Object) (admission.Warnings, error) {
+func (v *ClusterTemplateValidator) ValidateDelete(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	template, ok := obj.(*v1alpha1.ClusterTemplate)
+	if !ok {
+		return admission.Warnings{"Wrong object"}, apierrors.NewBadRequest(fmt.Sprintf("expected ClusterTemplate but got a %T", obj))
+	}
+
+	managedClusters := &v1alpha1.ManagedClusterList{}
```

Comment:
we should add a FieldIndexer to index over the `.spec.template` field instead of iterating over all existing clusters ( `mgr.GetFieldIndexer().IndexField()`)

Reply:
During the fixing of unit tests I've found a case where indexing will not work:
1. cluster template exists (because it is created first)
2. managed cluster doesn't exist (for any reason, i.e. ProviderTemplateReconciler fails or as in internal/controller/template_controller_test.go where only templates are created)
3. cluster template deletion will fail due to "Index with name field..." error

Reply:
You have this "Index with name field:.spec.template does not exist" error because you haven't added this indexer to the manager used in the tests.

Reply:
it's in `suite_test.go`
If you add the indexer here, you will no longer have this error in tests.